packages/graph/hash_graph/lib/graph/src/store/query/filter.rs

```diff
@@ -71,6 +74,49 @@ where
     }
 }
 
+impl<'q> Filter<'q, Entity> {
+    /// Creates a `Filter` to search for all entities at their latest version.
+    #[must_use]
+    pub const fn for_latest_entity_version() -> Self {
+        Self::Equal(
+            Some(FilterExpression::Path(EntityQueryPath::Version)),
+            Some(FilterExpression::Parameter(Parameter::Text(Cow::Borrowed(
+                "latest",
+            )))),
+        )
+    }
+
+    /// Creates a `Filter` to search for a specific entities at their latest version, identified by
+    /// its [`EntityId`].
+    #[must_use]
+    pub fn for_latest_entity_id(entity_id: EntityId) -> Self {
```

Comment:
Small nit: I don't know if the name of this method correctly captures the content. I would perhaps consider this `for latest entity by entity id`. Otherwise I read it to be `filtering the entity id of the latest entity`.

Reply:
I updated the newly added function names, what do you think?

Reply:
Perfect, thank you!